### ISSUE TITLE
Allow for group-based cross-validation objects from scikit-learn 

### ISSUE THREAD

**Issue by PaulZhutovsky** [ing-bank/probatus#181]
**Problem Description**
Probatus feature elimination (e.g. `ShapRFECV`) currently does not allow for cross-validation objects which take `groups` variables (e.g. [StratifiedGroupKFold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedGroupKFold.html))

**Desired Outcome**
It would be great if this feature could be implemented as those `groups` can be used to prevent data leakage in (e.g.) the case where multiple samples from the same customer are available and therefore should be either only in the training or the test set but not in both. 

**Solution Outline**
The fix to this _should_ be quite simple and can follow the implementation of `scikit-learn`'s [RFECV](https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/feature_selection/_rfe.py#L635): One would need to add a `groups` variable (default: `None`) to the `fit`/`fit_compute` methods of `ShapRFECV` and pass it through to [self.cv.split](https://github.com/ing-bank/probatus/blob/56afdc916d816fdc31fd4e71f8eb29413ff5a829/probatus/feature_elimination/feature_elimination.py#L549)
